.github/workflows/lint.yml

```diff
@@ -0,0 +1,35 @@
+name: Lint
+
+on:
+  workflow_dispatch:
+  push:
+    paths:
+      - '**/*.py'
+    branches:
+      - main
+  pull_request:
+    paths:
+      - '**/*.py'
+    branches:
+      - main
```

Comment:
what do we think about restricting this to the *package* materials, but expanding it to more branches?

I definitely want less non-library code in the repo moving forward, and linting is generally good, but ... seems like it might be an unnecessary headache for operations for the library-adjacent code, while letting too much mess accumulate in library for the operational branches that we want to merge back in.